sigal/video.py

```diff
@@ -75,21 +75,29 @@ def video_size(source):
     return x, y
 
 
-def generate_video(source, outname, size, options=None):
+def generate_video(source, outname, settings, options=None):
     """Video processor.
 
     :param source: path to a video
     :param outname: path to the generated video
-    :param size: size of the resized video `(width, height)`
+    :param settings: settings dict
     :param options: array of options passed to ffmpeg
 
     """
     logger = logging.getLogger(__name__)
 
+    if settings['use_orig']:
+        # Keep original filename
+        filename = os.path.split(source)[1]
+        outpath = os.path.split(outname)[0]
+        outname = os.path.join(outpath, filename)
+        utils.copy(source, outname, symlink=settings['orig_link'])
+        return outname
+
```

Comment:
Correct. I added it in _generate_video_ because the same is done in _generate_image_. But I'll move it to _process_video_.